bin/reth/src/debug_cmd/count_opcodes.rs

```diff
@@ -0,0 +1,406 @@
+//! `reth debug count-opcodes` comamnd. Tool to count opcodes occurrencies.
+use crate::primitives::Bytes;
+use clap::Parser;
+use comfy_table::{Cell, Row, Table};
+use itertools::Itertools;
+use reth_db::{open_db_read_only, tables};
+use reth_primitives::ChainSpecBuilder;
+use reth_provider::ProviderFactory;
+use reth_revm::interpreter::{opcode, OpCode};
+use std::{collections::HashMap, path::PathBuf, sync::Arc};
+
+macro_rules! opcodes_tuples_vec {
+    ($map:expr, $take: ident) => {
+        $map.iter()
+            .map(|(k, v)| {
+                let mut string = String::new();
+                (0..k.len())
+                    .for_each(|i| string.push_str(&format!("{} ", &opcode_or_invalid(k[i]))));
+                (string, *v)
+            })
+            .sorted_by(|a, b| b.1.cmp(&a.1))
+            .take($take)
+            .collect::<Vec<_>>()
+    };
+}
```

Comment:
`sorted_by` collects into a vec, sorts, then returns `vec.into_iter()`, then you collect again, classic itertools perf footgun :)
Would be better to
```rust
let x = map.iter().map().collect();
x.sort_by_key(); // a.1 b.1 here are usize so you can use `by_key`
&x[..$take]
```

Reply:
Nice! I opted for using `sort_unstable_by_key` which should not allocate at all: we don't care if low occurrencies opcodes tuples get mixed